app/src/main/java/com/example/bookwormadventuresdeluxe2/IsbnScanActivity.java

```diff
@@ -0,0 +1,85 @@
+package com.example.bookwormadventuresdeluxe2;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.zxing.integration.android.IntentIntegrator;
+import com.google.zxing.integration.android.IntentResult;
+
+// https://stackoverflow.com/questions/15249409/getting-error-in-barcode-scanner-in-android
+// Better way for later: https://www.youtube.com/watch?v=drH63NpSWyk
+public class IsbnScanActivity extends AppCompatActivity implements View.OnClickListener
+{
+    public static int LAUNCH_SCAN_ISBN = 42069;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_isbn_scan);
+    }
+
+    /**
+     * Clicked the "Scan" button
+     *
+     * @param v
+     */
+    @Override
+    public void onClick(View v)
+    {
+        IntentIntegrator integrator = new IntentIntegrator(
+                IsbnScanActivity.this);
+        integrator.setOrientationLocked(true);
+        integrator.setBeepEnabled(false);
+        integrator.initiateScan();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item)
+    {
+        // I added this so the back button in the app bar would just end this activity
+        // Not start a new MyBooksActivity
+        // https://stackoverflow.com/questions/14437745/how-to-override-action-bar-back-button-in-android
+        switch (item.getItemId())
+        {
+            case android.R.id.home:
+                Intent intent = new Intent();
+                intent.putExtra("isbn", "4206669");
```

Comment:
Is there a reason we are adding this to the bundle?

Reply:
yes bc I am a dumbass